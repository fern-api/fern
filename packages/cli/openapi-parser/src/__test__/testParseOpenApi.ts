import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse } from "../openapi/parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(fixtureName: string, openApiFilename: string, asyncApiFilename?: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const openApiIr = await parse({
                absolutePathToAsyncAPI,
                absolutePathToOpenAPI,
                absolutePathToOpenAPIOverrides: undefined,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
