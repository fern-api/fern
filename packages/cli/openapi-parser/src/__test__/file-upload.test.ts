import { testParseOpenAPI } from "./testParseOpenApi";

describe("file-upload", () => {
    testParseOpenAPI("file-upload", "openapi.yml");
});
