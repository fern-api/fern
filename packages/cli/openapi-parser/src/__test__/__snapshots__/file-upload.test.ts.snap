// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file-upload file-upload parse open api 1`] = `
{
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errorStatusCode": [],
      "examples": [],
      "generatedRequestName": "UploadFileRequest",
      "headers": [],
      "internal": null,
      "method": "POST",
      "operationId": "upload-file",
      "path": "/upload",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "description": null,
        "name": null,
        "properties": [
          {
            "description": null,
            "key": "file",
            "schema": {
              "isArray": false,
              "isOptional": false,
              "type": "file",
            },
          },
          {
            "description": null,
            "key": "maybe_file",
            "schema": {
              "isArray": false,
              "isOptional": true,
              "type": "file",
            },
          },
          {
            "description": null,
            "key": "file_list",
            "schema": {
              "isArray": true,
              "isOptional": false,
              "type": "file",
            },
          },
          {
            "description": null,
            "key": "maybe_file_list",
            "schema": {
              "isArray": true,
              "isOptional": true,
              "type": "file",
            },
          },
        ],
        "type": "multipart",
      },
      "requestNameOverride": null,
      "response": null,
      "sdkName": null,
      "server": [],
      "summary": "Upload a file",
      "tags": [],
    },
  ],
  "errors": {},
  "globalHeaders": null,
  "hasEndpointsMarkedInternal": false,
  "nonRequestReferencedSchemas": [],
  "schemas": {},
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Test file upload",
  "variables": {},
  "webhooks": [],
}
`;
